### PR TITLE
Build and test on ARM Android.

### PR DESCRIPTION
**Pull request by pietro**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -94,6 +96,30 @@ matrix:
...
+    - env: TARGET_X=arm-linux-androideabi CC_X=arm-linux-androideabi-gcc CXX_X=arm-linux-androideabi-g++ MODE_X=RELWITHDEBINFO KCOV=0
+      rust: stable
+      addons:
+        apt:
+          packages:
+            - expect
+            - openjdk-6-jre-headless
+
+    - env: TARGET_X=arm-linux-androideabi CC_X=arm-linux-androideabi-gcc CXX_X=arm-linux-androideabi-g++ MODE_X=DEBUG KCOV=0
+      rust: nightly
```
Comment (briansmith) [inline .travis.yml]: You are missing the `rust: nightly` + `MODE_X=RELWITHDEBINFO` option. Please add it.

File: mk/travis.sh [hunk 2]
```diff
@@ -61,7 +69,27 @@ rustc --version
...
+  emulator64-arm @arm-18 -no-audio -no-window &
+  adb wait-for-device
+  adb push target/$TARGET_X/debug/ring-* /data/ring-test
+  adb shell mkdir -p /data/crypto/poly1305
+  adb shell mkdir -p /data/crypto/cipher
+  adb shell mkdir -p /data/src
+  adb shell mkdir -p /data/third-party/NIST
+  adb push crypto/poly1305/poly1305_test.txt /data/crypto/poly1305/poly1305_test.txt
+  adb push crypto/cipher/test /data/crypto/cipher/test
+  for testfile in `ls src/*_tests.txt`; do adb push $testfile /data/$testfile ; done
```
Comment (briansmith) [inline mk/travis.sh]: Can't we do something like this?:
```
for testfile in `find src crypto -name "*_tests.txt"`; do
   adb shell mkdir -p `dirname $testfile`
   ....
````
?

**Comment by briansmith** [comment]
Please add the "I agree..." statement to the commit message (wrapped at 72 characters) and address the two nits.

This looks awesome!

File: .travis.yml (hunk 1 above)
Comment (pietro) [inline .travis.yml]: Added.

**Comment by pietro** [comment]
I added the wrapped  "I agree..." statement to the commit message. I didn't add it as a comment to the PR though.

**Comment by briansmith** [comment]
Unfortunately, the build fails:
```
+adb wait-for-device
+emulator64-arm @arm-18 -no-audio -no-window
emulator64-arm: error while loading shared libraries: libQt5Widgets.so.5: cannot open shared object file: No such file or directory
* daemon not running. starting it now on port 5037 *
* daemon started successfully *
```

**Comment by pietro** [comment]
Weird I haven't had that issue both locally and on travis. I'll do some more testing.

**Comment by pietro** [comment]
Sorry. I fixed the build and added a few things:

1. `travis_wait` to the android builds because the long running tests on release builds.
2. logic to copy the right binary (debug or release) to the emulator image.

@briansmith can you please review those? thanks.

**Comment by briansmith** [comment]
Awesome! Thanks! Landed as 72616bbf0580a059b844ffb42e6d8830073df56d.

**Pull request closed without merging**